addons/portal/controllers/portal.py

```diff
@@ -146,8 +146,11 @@ def _prepare_portal_layout_values(self):
         # get customer sales rep
         sales_user_sudo = request.env['res.users']
         partner_sudo = request.env.user.partner_id
+        fallback_sales_user = partner_sudo.commercial_partner_id.user_id
```

Comment:
Maybe we could only access that field in case the first condition is not met?

That way we avoid useless queries.

Reply:
@awa-odoo Agreed :)

I've done the change, it seems better now :)

Cheers